### PR TITLE
Roll release-next for 0.15 release

### PR DESCRIPTION
**Pull request by meyskens**
This prepares the testing for release-previous to be 0.15 and release-next to be 0.16

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -254,27 +254,7 @@ tide:
     - jetstack/cert-manager
     excludedBranches:
     - master
```
Comment (munnerz) [inline config/config.yaml:256]: We don't want to exclude any branches from this config else tide won't work on those branches. We only use `include/exclude` when code freeze is enabled in order to have different config per branch

File: config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml [hunk 2]
```diff
@@ -533,3 +533,60 @@ presubmits:
         options:
         - name: ndots
           value: "1"
+
+  - name: pull-cert-manager-e2e-v1-18
+    cluster: gke
+    context: pull-cert-manager-e2e-v1-18
+    always_run: true
```
Comment (munnerz) [inline config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml:540]: We don't want `always_run: true` along with `optional: true`

Comment (munnerz) [reply config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml:540]: Also: it looks like you didn't update this file by copying and pasting from `cert-manager-presubmits.yaml` in the parent directory as this same issue is still on the `v1-17` job? (I made the same mistake last time, which is why it's on that job and I think you've c+p'd that 😅)

File: config/config.yaml (hunk 1 above)
Comment (meyskens) [reply config/config.yaml:256]: I see!

File: config/config.yaml [hunk 3]
```diff
@@ -249,26 +249,6 @@ tide:
     - do-not-merge/release-note-label-needed
     - do-not-merge/work-in-progress
     - needs-rebase
-  # Maintain separate cert-manager configuration to make it easy to enable code freeze
```
Comment (munnerz) [inline config/config.yaml:252]: 👀 sorry I think we've got mixed up - this block *should* still exist, but you need to just remove the `excludedBranches` section from it as we have the same merge requirements for all branches in the repo when we aren't in code freeze

File: config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml [hunk 4]
```diff
@@ -537,8 +537,8 @@ presubmits:
   - name: pull-cert-manager-e2e-v1-18
     cluster: gke
     context: pull-cert-manager-e2e-v1-18
+    optional: false
```
Comment (munnerz) [inline config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml:540]: This is the 'previous' file - if we are switching the default Kubernetes e2e job from v1.17 to v1.18, we also need to update the branch protection config here: https://github.com/jetstack/testing/blob/a746b992b159beb6ddea486a3f383c060a4271d4/config/config.yaml#L36-L51

I think we'll want the same default across all branches come to think, unless we configure branch-specific merge protection config (which starts getting quite confusing..) 🙄

**Comment by munnerz** [comment]
Also need to update milestone_applier config: https://github.com/jetstack/testing/blob/2dec3b891bc246b5050fb79f11bd24dca8d1b4df/config/plugins.yaml#L84-L101

File: config/jobs/cert-manager/cert-manager-presubmits.yaml [hunk 5]
```diff
@@ -619,14 +619,14 @@ presubmits:
   - name: pull-cert-manager-e2e-v1-18
     cluster: gke
     context: pull-cert-manager-e2e-v1-18
-    always_run: false
-    optional: true
+    always_run: true
+    optional: false
```
Comment (munnerz) [inline config/jobs/cert-manager/cert-manager-presubmits.yaml:623]: You need to set `optional: true` and `always_run: false` on the v1.17 job too :)

File: config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml [hunk 6]
```diff
@@ -372,7 +372,7 @@ presubmits:
     agent: kubernetes
```
Comment (munnerz) [inline config/jobs/cert-manager/release-previous/cert-manager-release-previous-presubmits.yaml:372]: 2 lines up from here (https://github.com/jetstack/testing/pull/380/files#diff-8126d44f4ec7d37159faa0210dfb2e02R370) you've still got `always_run: true` on the v1.15 job from release-previous

**Pull request merged**